### PR TITLE
Chore!: Removed `WorkflowDef.prepare()` and `WorkflowRun.start()` functions from public API

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

.prepare and .start functions were bringing no value except making the WorkflowRun state-dependent

# This PR's solution

Remove those functions

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_base/_workflow.py [hunk 1]
```diff
@@ -158,42 +158,50 @@ def local_run(self) -> _R:
...
-        "Prepares" workflow for running. Call ".start()" on the result to
-        schedule the workflow for execution.
+        Schedules workflow for execution.
 
         Args:
-            config: SDK needs to know where to execute the workflow. The config
-                contains the required details. This can be a RuntimeConfig object, or
-                the name of a saved configuration.
+            config: SDK needs to know where to execute the workflow. This
+                objects contains the required details.
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_workflow.py:173]: **nitpick:** ‏I'm not sure what this comment change means

File: src/orquestra/sdk/_base/_workflow.py [hunk 2]
```diff
@@ -158,42 +158,50 @@ def local_run(self) -> _R:
...
-            ProjectInvalidError: when only 1 out of project and workspace is passed
         """
+        # This exists for users who have gotten used to doing `run()`. Once this has
+        # been released, the following release should make config a required argument
+        # and remove this check.
+        if config is None:
+            raise FutureWarning(
+                "Please specify the runtime configuration for this run. "
+                "The built in `local` and `in_process` configurations can be used by "
+                'calling `run.("local")` and `run("in_process")` respectively. '
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_workflow.py:191]:
```suggestion
                'calling `run("local")` and `run("in_process")` respectively. '
```

File: src/orquestra/sdk/_base/_workflow.py [hunk 3]
```diff
@@ -158,42 +158,50 @@ def local_run(self) -> _R:
...
             workspace_id: ID of the workspace for workflow - supported only on CE
             project_id: ID of the project for workflow - supported only on CE
 
         Raises:
-            ConfigNameNotFoundError: when the configuration has not been saved prior to
-                this point.
             orquestra.sdk.exceptions.DirtyGitRepo: (warning) when a task def used by
                 this workflow def has a "GitImport" and the git repo that contains it
                 has uncommitted changes.
-            ProjectInvalidError: when only 1 out of project and workspace is passed
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_workflow.py:187]: **question:** ‏What happens if I provide only a project?

**Review by jamesclark-Zapata — CHANGES_REQUESTED** [review]
Could you also remove any unused exceptions, like `WorkflowRunNotStarted`, from the exceptions module?

File: src/orquestra/sdk/_base/_workflow.py (hunk 3 above)
Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/_workflow.py:187]: Same as above - github shows it weirdly, although you are right we can get this exception in here, so Im adding the docs

**Comment by SebastianMorawiec** [comment]
> Could you also remove any unused exceptions, like `WorkflowRunNotStarted`, from the exceptions module?

This exception is still used in different place. Agreed to leave it

**Review by jamesclark-Zapata — APPROVED** [review]

**Pull request merged**